bundles/org.openhab.binding.metofficedatahub/src/main/java/org/openhab/binding/metofficedatahub/internal/MetOfficeDataHubSiteHandler.java

```diff
@@ -0,0 +1,587 @@
+                    getQuantityTypeState(data.getMaxScreenTemperature(), SIUnits.CELSIUS));
+
+            updateState(channelPrefix + SITE_HOURLY_FEELS_LIKE_TEMPERATURE,
+                    getQuantityTypeState(data.getFeelsLikeTemperature(), SIUnits.CELSIUS));
+
+            updateState(channelPrefix + SITE_HOURLY_SCREEN_RELATIVE_HUMIDITY,
+                    getQuantityTypeState(data.getScreenRelativeHumidity(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_HOURLY_VISIBILITY, getQuantityTypeState(data.getVisibility(), METRE));
+
+            updateState(channelPrefix + SITE_HOURLY_PROBABILITY_OF_PRECIPITATION,
+                    getQuantityTypeState(data.getProbOfPrecipitation(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_HOURLY_PRECIPITATION_RATE,
+                    getQuantityTypeState(data.getPrecipitationRate(), Units.MILLIMETRE_PER_HOUR));
+
+            updateState(channelPrefix + SITE_HOURLY_TOTAL_PRECIPITATION_AMOUNT,
+                    getQuantityTypeState(data.getTotalPrecipAmount(), MILLI(METRE)));
+
+            updateState(channelPrefix + SITE_HOURLY_TOTAL_SNOW_AMOUNT,
+                    getQuantityTypeState(data.getTotalSnowAmount(), MILLI(METRE)));
+
+            updateState(channelPrefix + SITE_HOURLY_PRESSURE, getQuantityTypeState(data.getPressure(), SIUnits.PASCAL));
+
+            updateState(channelPrefix + SITE_HOURLY_WIND_SPEED_10M,
+                    getQuantityTypeState(data.getWindSpeed10m(), Units.METRE_PER_SECOND));
+
+            updateState(channelPrefix + SITE_HOURLY_MAX_10M_WIND_GUST,
+                    getQuantityTypeState(data.getMax10mWindGust(), Units.METRE_PER_SECOND));
+
+            updateState(channelPrefix + SITE_HOURLY_WIND_GUST_SPEED_10M,
+                    getQuantityTypeState(data.getWindGustSpeed10m(), Units.METRE_PER_SECOND));
+
+            updateState(channelPrefix + SITE_HOURLY_SCREEN_DEW_POINT_TEMPERATURE,
+                    getQuantityTypeState(data.getScreenDewPointTemperature(), SIUnits.CELSIUS));
+
+            updateState(channelPrefix + SITE_HOURLY_UV_INDEX, getDecimalTypeState(data.getUvIndex()));
+
+            updateState(channelPrefix + SITE_HOURLY_WIND_DIRECTION_FROM_10M,
+                    getQuantityTypeState(data.getWindDirectionFrom10m(), Units.DEGREE_ANGLE));
+        }
+    }
+
+    public static String getLastHour() {
+        long timeRoundedToLastHour = System.currentTimeMillis();
+        timeRoundedToLastHour -= timeRoundedToLastHour % 3600000;
+        final Instant instant = new Date(timeRoundedToLastHour).toInstant();
+        return instant.toString().substring(0, 16) + "Z";
+    }
+
+    public static String getStartOfDay() {
+        long timeRoundedToDayStart = System.currentTimeMillis();
+        timeRoundedToDayStart -= timeRoundedToDayStart % 86400000;
+        final Instant instant = new Date(timeRoundedToDayStart).toInstant();
```

Comment:
Please check, but i think this is better.
```suggestion
        final Instant instant = Instant.now().truncatedTo(ChronoUnit.DAYS);
```

Reply:
That's much nicer - I was looking through those API's but missed that! Thank you - I've normalized the code to a common approach based on this :)